heron/healthmgr/src/java/com/twitter/heron/healthmgr/sensors/BaseSensor.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.healthmgr.sensors;
+
+import java.time.Duration;
+
+import com.microsoft.dhalion.api.ISensor;
+
+import com.twitter.heron.healthmgr.HealthPolicyConfig;
+import com.twitter.heron.healthmgr.HealthPolicyConfigReader.PolicyConfigKey;
+
+public abstract class BaseSensor implements ISensor {
+  static final Duration DEFAULT_METRIC_DURATION = Duration.ofSeconds(300);
+  static final String COMPONENT_STMGR = "__stmgr__";
+
+  enum MetricName {
```

Comment:
No, custom sensors will not need to add metric names to this enum. This enum simplifies development and testing of sensor implementation shipped with heron.